### DISCUSSION TITLE
Issue Reverting to Default HSS 0.99.29 on Video Kit After Update

### DISCUSSION THREAD

**Discussion by JustinAlyne** [polarfire-soc/polarfire-soc-documentation#403] [How to]
Hello,

I am seeking assistance with reverting the HSS on my video kit to its default version. Here are the details:
Development board: https://www.microchip.com/en-us/development-tool/MPFS250-VIDEO-KIT 
Initial Upgrade: Successfully upgraded from the default HSS 0.99.29 to 0.99.37 using Flashpro 5 with the MPFS_VIDEO_KIT_BASE_DESIGN_2023_10.job file.

- Attempted Downgrade: The earliest available HSS version is 0.99.33 from the [v2022.09](https://github.com/polarfire-soc/polarfire-soc-video-kit-reference-design/releases/tag/v2022.09). Attempting to use this resulted in an error indicating a mismatch with the expected device: "Error: programmer '11037401': Device 1: Found: MPFS250(T|TS|TL|TLS), Expected: MPFS250T_ES".

- Note: MPFS250T_ES seems to be for the Icicle Kit, not the Video Kit, yet it is listed under the Video Kit releases.
- Reason for Downgrade: The board originally shipped with HSS 0.99.29 and a corresponding Yocto Linux image, both of which worked perfectly with the HDMI TX demo. 
- Since updating to HSS 0.99.37 and Yocto image [v2024.02](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/tag/v2024.02), the HDMI output no longer functions with the IMX334 video demo.
+ With HSS 0.99.29: The IMX334 camera and HDMI TX worked together without issue.
+ With HSS 0.99.37 and Yocto 2024.02: The IMX334 camera works (verified via the web demo), but not when running the HDMI TX demo. 

Support:
Is there a repository or archive where I can access the original HSS 0.99.29?
Could there be compatibility issues with the new HSS/Yocto versions affecting HDMI output, and how might I resolve them?

Any guidance or resources would be greatly appreciated.
-Justin

**Comment by hughbreslin** [comment]
Hi @JustinAlyne how did you update your HSS - was it by itself (i.e via softconsole) or as part of a design update (i.e reference design + HSS client) either through Libero or FPExpress? If its the design update option I think the issue is actually the hardware configuration change has caused your issue and not the HSS change. The HSS is just embedded software and shouldn't have an influence on the HDMI working or not

**Reply by JustinAlyne** [reply]
Hi @hughbreslin I did a design update through the FPExpress. I should've mention this more clearer but essentially I upgraded the HSS to the latest version which no longer works with the default linux image that is on my eMMC. However I flashed a MicroSD card with the Yocto image [v2024.02](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/tag/v2024.02) that has new device trees and configurations that I believe may have changed something such that the kernel no longer is able to communicate with the HDMI interface. I know that the v2024.02 yocto image still works with the camera because I'm able to run the webdemo but not able to use HDMI TX interface. 

But essentially it would help for me to debug and compare the configuration difference between the default Linux image on the board and the newest version aka v2024.02 

So if i were to downgrade to the default HSS my eMMC linux image would be compatible and boot up again. At the moment I'm not able to experiment and discover what the differences are because I can't revert my HSS back to the default state. So my current state is HSS version 0.99.37 with a Yocto image [v2024.02](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/tag/v2024.02). 

Is there a way for me to revert back to the default HSS for the Polarfire Video Kit?

**Reply by sc-mchp** [reply]
This is not an HSS issue, the GitHub reference designs do not support HDMI Tx, whereas the OOBE design includes HDMI Tx.

**Reply by JustinAlyne** [reply]
@shravanI35088 Was there a reason as to why it was discontinued? I would like to utilize the 4k IMX334 dual sensor cameras to run 3840x2160 but the webdemo only goes up to 1920x1072 and if I'm understanding you correctly the HDMI Tx is no longer supported with the latest GitHub reference designs?

**Reply by sc-mchp** [reply]
As of now, the demo will work with 2K resolution, and GitHub HDMI transmission may be available in the future. If you want out-of-box (OOB) images, you can obtain them through tech support.

Thanks,
Shravan
